src/components/EditCoursePage/EditCourseForm.test.jsx

```diff
@@ -244,6 +245,72 @@ describe('BaseEditCourseForm', () => {
     expect(disabledFields).toHaveLength(1);
   });
 
+  it('checks preview url renders correctly for old url slug format', () => {
+    const urlSlug = 'test-url-slug';
+    const MARKETING_SITE_PREVIEW_URL_ROOT = process.env.MARKETING_SITE_PREVIEW_URL_ROOT;
+    const courseInfoWithUrlSlug = { ...courseInfo, data: { ...courseInfo.data, url_slug: urlSlug } };
+    const wrapper = shallow(<BaseEditCourseForm
+      handleSubmit={() => null}
+      title={initialValuesFull.title}
+      initialValues={initialValuesFull}
+      currentFormValues={initialValuesFull}
+      number="Test101x"
+      entitlement={{ sku: 'ABC1234' }}
+      courseStatuses={[REVIEWED]}
+      courseInfo={courseInfoWithUrlSlug}
+      courseOptions={courseOptions}
+      courseRunOptions={courseRunOptions}
+      uuid={initialValuesFull.uuid}
+      type={initialValuesFull.type}
+      id="edit-course-form"
+    />);
+    expect(wrapper.instance().getLinkComponent([REVIEWED], courseInfoWithUrlSlug)).toEqual(
+      <>
+        <Hyperlink
+          name="preview-url"
+          destination={`${MARKETING_SITE_PREVIEW_URL_ROOT}/course/${urlSlug}`}
+          target="_blank"
+        >
+          View Preview Page
+        </Hyperlink>
+        <span className="d-block">Any changes will go live when the website next builds</span>
+      </>,
+    );
+  });
+
+  it('checks preview url renders correctly for new slug format', () => {
+    const urlSlug = 'learn/test-url-slug/test';
+    const MARKETING_SITE_PREVIEW_URL_ROOT = process.env.MARKETING_SITE_PREVIEW_URL_ROOT;
+    const courseInfoWithUrlSlug = { ...courseInfo, data: { ...courseInfo.data, url_slug: urlSlug } };
+    const wrapper = shallow(<BaseEditCourseForm
+      handleSubmit={() => null}
+      title={initialValuesFull.title}
+      initialValues={initialValuesFull}
+      currentFormValues={initialValuesFull}
+      number="Test101x"
+      entitlement={{ sku: 'ABC1234' }}
+      courseStatuses={[REVIEWED]}
+      courseInfo={courseInfoWithUrlSlug}
+      courseOptions={courseOptions}
+      courseRunOptions={courseRunOptions}
+      uuid={initialValuesFull.uuid}
+      type={initialValuesFull.type}
+      id="edit-course-form"
+    />);
+    expect(wrapper.instance().getLinkComponent([REVIEWED], courseInfoWithUrlSlug)).toEqual(
```

Comment:
You wish to assert something about `wrapper`. You do that by calling a method(`getLinkComponent`) on `wrapper` that does not depend on `wrapper` but merely on its arguments. So essentially `wrapper` is irrelevant in this test and you are just testing `getLinkComponent`. If this is indeed the case we should remove `wrapper` and call `getLinkComponent` directly?

Reply:
It's a class method and can't be called independently, IMO.
> In React, class methods are typically bound to component instances and are meant to be invoked as part of the component lifecycle or in response to specific events. They are not designed to be tested independently like regular functions.

Reply:
Hmm.
I feel it is better to restructure these tests to use something like `wrapper.find` to get the `Hyperlink` and then assert on the `props`, rather than calling `getLinkComponent`.

Reply:
I would suggest writing these tests with rtl instead of enzmye. Enzyme is going to be out of fashion, so let's move to RTL with a new code instead of doing that work later.

Reply:
And I also 2nd @zawan-ila  that rather than asserting on entire component, verify its rendering and props value that are important in this unit test.

Reply:
> I would suggest writing these tests with rtl instead of enzmye. Enzyme is going to be out of fashion, so let's move to RTL with a new code instead of doing that work later.

Testing Redux Form in RTL can be more complex compared to enzyme. It requires managing the store and providers to maintain the form's state during testing. Additionally, moving a few tests from enzyme to RTL may introduce confusion. It might be better to address this in a follow-up pull request where we can focus on properly integrating RTL with Redux Form.

Reply:
What's the complexity value? Any links/refs on that? If it it indeed high effort, we can do it later. But if it is quick,  lets start from this PR.

Reply:
We need to setup Stores and Providers for the tests just like in the following links. 

Ref Links:
https://stackoverflow.com/questions/46932791/redux-form-test-using-jest
https://dev.to/ip4422/redux-form-and-typescript-testing-with-react-testing-library-5bla